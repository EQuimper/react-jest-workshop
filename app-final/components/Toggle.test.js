import React from 'react'
import renderer from 'react-test-renderer'

import Toggle from './Toggle'

test('has toggle--off class applied by default', () => {
  throw new Error('this error is on line 7 of Toggle.test.js which is transpiled for tests!')
  snapshotProps()
})

test('has toggle--on class applied when initialToggledOn specified to true', () => {
  snapshotProps({initialToggledOn: true})
})

function snapshotProps(props) {
  const component = renderToggle(props)
  expect(component).toMatchSnapshot()
}

/**
 * Render the <Toggle /> component with enzyme and return the wrapper
 * @param {Object} props={} - the props to apply to the <Toggle /> component
 * @return {Wrapper} - the enzyme wrapper
 */
function renderToggle(props) {
  return renderer.create(<Toggle {...getProps(props)} />)
}

function getProps(props = {}) {
  return {
    onToggle() {},
    children: 'Toggle Me',
    ...props,
  }
}
