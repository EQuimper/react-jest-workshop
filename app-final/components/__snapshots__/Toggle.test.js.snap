exports[`test should have toggle--off class applied by default 1`] = `
<div
  className="toggle toggle--off">
  <button
    onClick={[Function anonymous]}>
    Toggle Me
  </button>
</div>
`;

exports[`test should have toggle--on class applied when initialToggledOn specified to true 1`] = `
<div
  className="toggle toggle--on">
  <button
    onClick={[Function anonymous]}>
    Toggle Me
  </button>
</div>
`;
